extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java

```diff
@@ -0,0 +1,284 @@
+        KafkaDevServicesBuildTimeConfig devServicesConfig = cfg.devservices;
+        boolean devServicesEnabled = devServicesConfig.enabled.orElse(true);
+        return new KafkaDevServiceCfg(devServicesEnabled,
+                devServicesConfig.imageName,
+                devServicesConfig.port.orElse(0));
+    }
+
+    private static class KafkaBroker {
+        private final String url;
+        private final Closeable closeable;
+
+        public KafkaBroker(String url, Closeable closeable) {
+            this.url = url;
+            this.closeable = closeable;
+        }
+
+        public String getBootstrapServers() {
+            return url;
+        }
+
+        public Closeable getCloseable() {
+            return closeable;
+        }
+    }
+
+    private static final class KafkaDevServiceCfg {
+        private final boolean devServicesEnabled;
+        private final String imageName;
+        private final Integer fixedExposedPort;
+
+        public KafkaDevServiceCfg(boolean devServicesEnabled, String imageName, Integer fixedExposedPort) {
+            this.devServicesEnabled = devServicesEnabled;
+            this.imageName = imageName;
+            this.fixedExposedPort = fixedExposedPort;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            KafkaDevServiceCfg that = (KafkaDevServiceCfg) o;
+            return devServicesEnabled == that.devServicesEnabled && Objects.equals(imageName, that.imageName)
+                    && Objects.equals(fixedExposedPort, that.fixedExposedPort);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(devServicesEnabled, imageName, fixedExposedPort);
+        }
+    }
+
+    /**
+     * Container configuring and starting the Red Panda broker.
+     * See https://vectorized.io/docs/quick-start-docker/
+     */
+    private static final class RedPandaKafkaContainer extends GenericContainer<RedPandaKafkaContainer> {
+
+        private final int port;
+
+        private static final String STARTER_SCRIPT = "/redpanda.sh";
+
+        private RedPandaKafkaContainer(DockerImageName dockerImageName, int fixedExposedPort) {
+            super(dockerImageName);
+            this.port = fixedExposedPort;
+            withNetwork(Network.SHARED);
+            withExposedPorts(KAFKA_PORT);
+            // For red panda, we need to start the broker - see https://vectorized.io/docs/quick-start-docker/
+            if (dockerImageName.getRepository().equals("vectorized/redpanda")) {
+                withCreateContainerCmdModifier(cmd -> {
+                    cmd.withEntrypoint("sh");
+                });
+                withCommand("-c", "while [ ! -f " + STARTER_SCRIPT + " ]; do sleep 0.1; done; " + STARTER_SCRIPT);
```

Comment:
Given the starter script is just a single command, wouldn't it be better to just define it here? Something like:

```java
withCommand("-c", "/usr/bin/rpk redpanda start ...");
```

?

(Perhaps it doesn't even have to be executed in shell, but I don't really know.)

Reply:
Ah sorry, stupid question. At this point, you don't know the mapped port. Please ignore :-)

Reply:
I need to try, I reused the approach from Test Container.

Reply:
Nah, it needs to be like this, I asked before thinking :-)